Polygon_mesh_processing/include/CGAL/Side_of_triangle_mesh.h

```diff
@@ -195,8 +203,21 @@ class Side_of_triangle_mesh
    */
   Bounded_side operator()(const Point& point) const
   {
+    bool is_outside = tree_built;
+    if(!is_outside){
+      is_outside = (point.x() < box.xmin()
+                    || point.x() > box.xmax()
+                    || point.y() < box.ymin()
+                    || point.y() > box.ymax()
+                    || point.z() < box.zmin()
+                    || point.z() > box.zmax());
+      if(is_outside)
+        return CGAL::ON_UNBOUNDED_SIDE;
+    }
+
+    tree_built = true;
     return internal::Point_inside_vertical_ray_cast<GeomTraits, AABB_tree>()(
-      point, *tree_ptr, ray_functor, vector_functor);
+          point, *tree_ptr, ray_functor, vector_functor);
```

Comment:
That call (in the return) triggers a call on the AABB_tree, that is then built lazily.

Reply:
yes but `tree_built` has nothing to do with it so it was useless. And the previous patch is not needed IMO. When I'm saying lazy construction, I also mean lazy allocation/creation of the AABB-tree. But it requires some tricks since the AABB-tree might not be built from a mesh (so it might not even be possible).

Reply:
Why is it faster like this then ?

Reply:
I don't see any use of the value of `tree_built` that is meaningful. Disabling the check using the bbox is not going to speed anything significantly.

Reply:
So it's better to perform systematically one to six tests than to perform one ?